### PR TITLE
Cross compilation builds

### PR DESCRIPTION
**Pull request by paulidale**
Add some cross compiling builds to test things aren't broken.

Primarily aimed at the assembly but there are other places things go architecture specific.

### CODE REVIEW

File: .github/workflows/cross-compiles.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+jobs:
+  cross-compilation:
+    strategy:
+      fail-fast: false
+      matrix:
+        # The platform matrix strings are two fields separated with a colon.
+        # The first field is the package to install using apt.
+        # The second field is passed to Configure as command line arguments.
+        platform: [
+          'gcc-aarch64-linux-gnu:-cross-compile-prefix=aarch64-linux-gnu- linux-aarch64',
```
Comment (levitte) [inline .github/workflows/cross-compiles.yml:14]: A dash is missing in `-cross-compilation`

File: .github/workflows/cross-compiles.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+          'gcc-alpha-linux-gnu:--cross-compile-prefix=alpha-linux-gnu- linux-alpha-gcc',
+          'gcc-arm-linux-gnueabi:--cross-compile-prefix=arm-linux-gnueabi- linux-armv4',
+          'gcc-arm-linux-gnueabihf:--cross-compile-prefix=arm-linux-gnueabihf- linux-armv4',
+          'gcc-mips-linux-gnu:--cross-compile-prefix=mips-linux-gnu- linux-mips32',
+          'gcc-mipsel-linux-gnu:--cross-compile-prefix=mipsel-linux-gnu- linux-mips32',
+          'gcc-powerpc64le-linux-gnu:--cross-compile-prefix=powerpc64le-linux-gnu- linux-ppc64le',
+          'gcc-riscv64-linux-gnu:--cross-compile-prefix=riscv64-linux-gnu- linux64-riscv64',
+          'gcc-s390x-linux-gnu:--cross-compile-prefix=s390x-linux-gnu- linux64-s390x',
+          'gcc-sparc64-linux-gnu:--cross-compile-prefix=sparc64-linux-gnu- linux64-sparcv9',
+        ]
```
Comment (levitte) [inline .github/workflows/cross-compiles.yml:24]: I believe that if you make each line dictionaries, you won't have to play with `cut`:

``` yaml
      platform: [
        { target: gcc-aarch64-linux-gnu, config: '--cross-compile-prefix=aarch64-linux-gnu- linux-aarch64' },
        ...
```

Access with `${{ matrix.platform.target }}` and `${{ matrix.platform.config }}`

Might be worth trying out at least?

File: .github/workflows/cross-compiles.yml (hunk 1 above)
Comment (paulidale) [reply .github/workflows/cross-compiles.yml:14]: Hmmmm.... how did this pass?

File: .github/workflows/cross-compiles.yml (hunk 2 above)
Comment (paulidale) [reply .github/workflows/cross-compiles.yml:55]: Definitely.  Update pushed....
Much nicer.

File: .github/workflows/cross-compiles.yml (hunk 1 above)
Comment (paulidale) [reply .github/workflows/cross-compiles.yml]: It wasn't buliding that's why

File: .github/workflows/cross-compiles.yml [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+on: [pull_request, push]
+
+jobs:
+  cross-compilation:
+    strategy:
+      fail-fast: false
+      matrix:
+        # The platform matrix strings are two fields separated with a colon.
+        # The first field is the package to install using apt.
+        # The second field is passed to Configure as command line arguments.
```
Comment (levitte) [inline .github/workflows/cross-compiles.yml:12]: You might want to edit this comment :wink:

Comment (paulidale) [reply .github/workflows/cross-compiles.yml]: Done.

**Review by levitte — APPROVED** [review]

**Comment by paulidale** [comment]
Two of the builds were failing, I've added fixes to get them going again.  A new review will be required.

File: crypto/ppccap.c [hunk 4]
```diff
@@ -68,11 +68,12 @@ int bn_mul_mont(BN_ULONG *rp, const BN_ULONG *ap, const BN_ULONG *bp,
      * no opportunity to figure it out...
      */
 
-    if (num == 6)
+    if (num == 6) {
```
Comment (slontis) [inline crypto/ppccap.c:71]: Is it really that stuupid?

Comment (paulidale) [reply crypto/ppccap.c:71]: Just a hint saying the else could be misinterpreted.

**Review by slontis — APPROVED** [review]

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master.

**Pull request closed without merging**